models.py

```diff
@@ -70,6 +69,17 @@ def prev(self):
       .order_by(Datasets.created_at)\
       .first()
 
+  def items(self):
+    return Datasets.query\
+      .filter(Datasets.study_id==self.study_id)\
+      .order_by(Datasets.created_at)\
+      .all()
+
+  def study_name(self):
```

Comment:
This method isn't necessary at all.

The Datasets model already has a "study" property as a backred from Studies

  datasets = db.relationship('Datasets', lazy="dynamic", backref="study")

So all you need to do is...

dataset.study.name

to get the name. The ORM generates the query for you.